pandas/io/common.py

```diff
@@ -135,7 +142,7 @@ def get_filepath_or_buffer(filepath_or_buffer, encoding=None):
 
     if _is_url(filepath_or_buffer):
         req = _urlopen(str(filepath_or_buffer))
```

Comment:
Yeah, that's a great idea.  I couldn't set a Content-Encoding header on a file in github, but I can in s3, so I'll add that code in and a new test.